### PR TITLE
Don't expose loadPyodide as global variable from ES6 module

### PR DESCRIPTION
**Pull request by hoodmane**

### CODE REVIEW

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane, LGTM and sorry for the late review. Please update the changelog.

**Comment by leopsidom** [comment]
Ideally it feels more natural to let rollup handle different target with the same input. But since we have a preference to use `loadPyodide` as global, this LGTM.

**Review by leopsidom — APPROVED** [review]

**Comment by hoodmane** [comment]
> Ideally it feels more natural to let rollup handle different target with the same input. But since we have a preference to use loadPyodide as global, this LGTM.

Can you explain what you mean? I think we discussed this before but I'm still not sure what the alternative looks like.

**Comment by hoodmane** [comment]
In particular, we want the non-es6 `pyodide.js` to be loadable in a `<script>` tag, where I think there is no option except to attach `loadPyodide` as a global?

File: src/js/pyodide.umd.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { loadPyodide } from "./pyodide";
+export { loadPyodide };
```
[the remark this answers is not in the document]
Comment (hoodmane) [reply src/js/pyodide.umd.ts:2]: I'm not even sure what `export` does for umd, but this is redundant.

**Comment by leopsidom** [comment]
> Can you explain what you mean? I think we discussed this before but I'm still not sure what the alternative looks like.

Yeah, it might be a while ago. But I think if default to rollup config, the alternative is to give the global object a name such as `pyodide` and then reference `loadPyodide` from it as `pyodide.loadPyodide`.

**Comment by leopsidom** [comment]
> In particular, we want the non-es6 `pyodide.js` to be loadable in a `<script>` tag, where I think there is no option except to attach `loadPyodide` as a global?

With current implementation,  it's possible to use `pyodide.loadPyodide` if we do not attach `loadPyodide` to global. But we need to change the name in [rollup config](https://github.com/pyodide/pyodide/blob/main/src/js/rollup.config.js#L13) to `pyodide`.

**Comment by leopsidom** [comment]
But since you are looking to fix this issue: https://github.com/pyodide/pyodide/issues/2255. It's likely there will be some changes made to the API. So it makes more sense to consolidate it over there.

**Comment by hoodmane** [comment]
Right, well as I said before, I don't love having to call it `pyodide.loadPyodide`. Maybe we should just leave the global variable in the umd file and encourage people to use the es6 module.

**Comment by hoodmane** [comment]
Thanks for the reviews @leopsidom and @ryanking13!

**Pull request merged**